src/utils.jl

```diff
@@ -82,11 +82,10 @@ function colwise_dot!(r::AbstractArray, a::AbstractMatrix, b::AbstractMatrix)
 end
 
 function colwise_sumsq!(r::AbstractArray, a::AbstractMatrix, c::Real)
-    n = length(r)
-    @check_argdims n == size(a, 2)
-    for j = 1:n
+    @check_argdims(eachindex(r) == axes(a, 2))
+    for j in axes(a, 2)
         v = zero(eltype(a))
-        @simd for i = 1:size(a, 1)
+        @simd for i in axes(a, 1)
             @inbounds v += abs2(a[i, j])
         end
         r[j] = v*c
```

Comment:
Maybe just change to
for (i, ai) in zip(eachindex(r), eachcol(a))
r[i] = sum(abs2, ai) * c
end
?

Reply:
I think out of scope for this PR.
I am avoiding changes to logic except in the one case where I can delete a method entirely.